backend/common/src/either_sink.rs

```diff
@@ -0,0 +1,57 @@
+use futures::sink::Sink;
+use pin_project_lite::pin_project;
+
+pin_project! {
+    #[project = EitherSinkInner]
+    pub enum EitherSink<A, B> {
+        A { #[pin] inner: A },
+        B { #[pin] inner: B }
+    }
+}
+
+/// A simple enum that delegates implementation to one of
+/// the two possible sinks contained within.
+impl <A, B> EitherSink<A, B> {
```

Comment:
formatting seems a bit of here; I think we are using `rustfmt` in this repo right?

but maybe not enforced by CI I guess.

Reply:
Yup; I'm using rustfmt but indeed I don't think CI enforces it at the mo. It does sometimes make the formatting a little "wierd" IMO but oh well :)